src/System.Reflection.Metadata/src/System/Reflection/Metadata/MetadataReader.cs

```diff
@@ -96,7 +95,7 @@ public unsafe MetadataReader(byte* metadata, int length, MetadataReaderOptions o
             this.Block = new MemoryBlock(metadata, length);
 
             _options = options;
-            this.Utf8Decoder = utf8Decoder;
+            this.StringDecoder = utf8Decoder;
```

Comment:
If we need to decode using another encoding (utf16, eg.) I assumed we'd just add another virtual method to MetadataStringDecoder rather than adding a new decoder class. But I guess the class has an Encoding on it... hmm.